components/base/Button/Button.tsx

```diff
@@ -0,0 +1,41 @@
+type TButtonSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+type TButtonVariation = 'default' | 'outlined';
+
+const buttonSizes: Record<TButtonSize, string> = {
+  sm: 'text-sm px-[14px] py-2',
+  md: 'text-sm px-4 py-[10px]',
+  lg: 'px-[18px] py-[10px]',
+  xl: 'px-5 py-3',
+  '2xl': 'text-lg px-7 py-4'
+};
+
+const buttonBase = 'font-semibold rounded-lg shadow-sm shadow-violet-100/5';
+
+const buttonStyles: Record<TButtonVariation, string> = {
+  default: `${buttonBase} text-white bg-violet-600 hover:bg-violet-700 focus:ring-4 ring-violet-100 border border-violet-600`,
+  outlined: `${buttonBase} text-black hover:border-gray-400 bg-white focus:ring-4 ring-gray-100 border border-gray-300`
+};
+
+const disabledButtonStyles: Record<TButtonVariation, string> = {
+  default: `${buttonBase} text-white bg-violet-200 hover:bg-violet-200 cursor-not-allowed border border-violet-200`,
+  outlined: `${buttonBase} text-black bg-gray-200 hover:bg-gray-200 border border-gray-300 cursor-not-allowed`
+};
+
+// TODO: Icon
+interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  size?: TButtonSize;
+  variation?: TButtonVariation;
+}
+
+const Button = ({ className, ...props }: Props) => {
+  const size = buttonSizes[props.size ?? 'md'];
+  let style = props.disabled
+    ? disabledButtonStyles[props.variation ?? 'default']
+    : buttonStyles[props.variation ?? 'default'];
+
```

Comment:
const isDisabled = props.disabled;
 const variation = props.variation ?? 'default'
 let style = isDisabled ? disabledButtonStyles[variation] : buttonStyles[variation]

Reply:
Wouldn't it be better to destroy the disabled prop from the props instead of defining it in a separate variable? @eraygundogmus 

```js
const Button = ({ className, disabled = false, ...props }: Props) => {

const variation = props.variation ?? 'default'
let style = disabled ? disabledButtonStyles[variation] : buttonStyles[variation]
```

Reply:
> Wouldn't it be better to destroy the disabled prop from the props instead of defining it in a separate variable? @eraygundogmus
> 
> ```js
> const Button = ({ className, disabled = false, ...props }: Props) => {
> 
> const variation = props.variation ?? 'default'
> let style = disabled ? disabledButtonStyles[variation] : buttonStyles[variation]
> ```

yes, guess that's even better

Reply:
Unless component specified as `disabled` it would be `undefined` like many other props inherited from html, I don't see any additional benefit from that, just discards `props` keyword before disabled. 

Same can be also applied to variation. 
```ts
const Button = ({ className, variation = 'default', ...props }: Props) => {
```

Reply:
@wralith my point was simplifying the code reability, nothing technical. 
`
let something = isSomething 
`
is more semantic than 
`
let something = else.isSomething 
`